secure/key-manager/src/libra_interface.rs

```diff
@@ -0,0 +1,217 @@
+                Error::DataDoesNotExist(format!(
+                    "LibraTimestampResource not found for account: {:?}",
+                    account
+                ))
+            }),
+            e => Err(Error::UnknownError(format!("{:?}", e))),
+        }
+    }
+
+    /// Submits a signed transaction to the Libra blockchain.
+    fn submit_signed_transaction(
+        &self,
+        signed_transaction: SignedTransaction,
+    ) -> Result<(), Error> {
+        self.client
+            .submit_signed_transaction(signed_transaction)
+            .map_err(|e| {
+                Error::UnknownError(format!(
+                    "Failed to submit signed transaction. Error: {:?}",
+                    e,
+                ))
+            })
+    }
+
+    /// Fetches the AccountState associated with a specific account.
+    pub(crate) fn get_account_state(&self, account: AccountAddress) -> Result<AccountState, Error> {
+        self.client.get_account_state(account, None).map_err(|e| {
+            Error::UnknownError(format!(
+                "Failed to get AccountState for account: {:?}. Error: {:?}",
+                account, e
+            ))
+        })
+    }
+}
+
+impl LibraInterface for JsonRpcLibraInterface {
+    fn libra_timestamp(&self) -> Result<u64, Error> {
+        let account = account_config::association_address();
+        let libra_timestamp_resource = self.get_libra_timestamp_resource(account)?;
+        Ok(libra_timestamp_resource.libra_timestamp.microseconds)
+    }
+
+    fn last_reconfiguration(&self) -> Result<u64, Error> {
+        let account = account_config::association_address();
+        let configuration_resource = self.get_configuration_resource(account)?;
+        Ok(configuration_resource.last_reconfiguration_time())
+    }
+
+    fn retrieve_sequence_number(&self, account: AccountAddress) -> Result<u64, Error> {
+        let account_resource = self.get_account_resource(account)?;
+        Ok(account_resource.sequence_number())
+    }
+
+    fn submit_transaction(&self, transaction: Transaction) -> Result<(), Error> {
+        if let Transaction::UserTransaction(signed_txn) = transaction {
+            self.submit_signed_transaction(signed_txn)
+        } else {
+            Err(Error::UnknownError(format!(
+                "Unable to submit a transaction type that is not a SignedTransaction: {:?}",
+                transaction
+            )))
+        }
+    }
+
+    fn retrieve_validator_config(&self, account: AccountAddress) -> Result<ValidatorConfig, Error> {
+        let validator_config_resource = self.get_validator_config_resource(account)?;
+        Ok(validator_config_resource.validator_config)
+    }
+
+    fn retrieve_validator_info(&self, account: AccountAddress) -> Result<ValidatorInfo, Error> {
+        let validator_set_account = account_config::validator_set_address();
+        let validator_set = self.get_validator_set(validator_set_account)?;
```

Comment:
should we merge this into here?

Reply:
Done! Personally, I prefer the split (it makes the code easier to re-use and modify), but I don't feel strongly!